components/style/stylist.rs

```diff
@@ -98,6 +101,16 @@ pub struct Stylist {
 
     /// The total number of times the stylist has been rebuilt.
     num_rebuilds: usize,
+
+    /// The set of registered custom property associated with the document.
+    /// Should be initialized as soon as possible.
+    #[cfg_attr(feature = "servo", ignore_heap_size_of = "Arc")]
+    registered_property_set: Option<Arc<RwLock<RegisteredPropertySet>>>,
```

Comment:
FWIW what we do for stylesheets and similar is keeping two sets, one on layout and one on script (see the `Document::stylesheets` field). That makes the handling of them easier from style, so we may want to consider that, given Gecko wouldn't have this annoying problem of having to move the set across threads.

In any case, I haven't gone through it in detail, but I'd be very confused if we'd need to mutate this kind of global state each time we style an element, so I suspect we can at least get rid of the `RwLock`.

Also, I see no call to `set_registered_custom_properties` in Stylo, right?